### PR TITLE
Add debugging with devTools relative cases and package and distribute…

### PR DESCRIPTION
**Pull request by xiaoyu3x**
| | Pass | Fail | Block|
---|---|---|---
Total |  6 | 0 | 0
Auto|  1 | 0 | 0
Manual | 5 | 0| 0

### CODE REVIEW

File: nwjs-iot/nwjsapi/debugging-with-devTools/http-server.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/http-server.py]: All the files(except README.md, *.json, ..) should have license.

File: nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ARE DISCLAIMED. IN NO EVENT SHALL INTEL CORPORATION BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+<html>
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html]: <html>, <head>, <body> labels are unnecessary. See https://github.com/crosswalk-project/crosswalk-test-suite/blob/master/tools/template/script_test_case.html.

File: nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+</head>
+<body id="demo">
+<ol>
+   <li>'Right Click App'</li>
+   <li>'Select "Inspect Background Page" '</li>
+   <li>'It will is pass if debugging Tools is opened '</li>
+</ol>
+<script>
+var events=require('events');
+var eventEmitter=new events.EventEmitter();
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html]: What are these events used for? Please add space between "=", "," "{" and etc.
And only need two whitespace when line feed.

File: nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
+import time
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.py]: Add Copyright.

Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.py]: Why you locate this util file under this folder? Isn't it a common util?

File: nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        return ret
+    else:
+        pgrep = subprocess.Popen(['pgrep', '-P', str(driver.service.process.pid)], stdout=subprocess.PIPE)
+        out, err = pgrep.communicate()
+        ret = (pgrep.returncode == 1)
+        if not ret and print_if_fail:
+            print 'live chrome processes:\n%s' % out
+            print 'pgrep exit with %s' % pgrep.returncode
+        # expect exit 1 from pgrep, which means no chrome process alive
+        return ret
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.py]: nwjs-iot/nwjsapi/debugging-with-devTools/nw_util.pyc
this is a temporary file, don't submit it.

File: nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html]: Doesn't it need "--remote-debug-...." command to launch this app?

File: nwjs-iot/nwjsapi/debugging-with-devTools/open-dev-tools/test.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/open-dev-tools/test.py]: Add Copyright.

File: nwjs-iot/nwjsapi/debugging-with-devTools/open-dev-tools/test.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)),'..'))
+from nw_util import *
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+
+chrome_options=Options()
+chrome_options.add_argument("nwapp="+os.path.dirname(os.path.realpath(__file__)))
+
+driver=webdriver.Chrome(executable_path=os.environ["CHROMEDRIVER"],chrome_options=chrome_options)
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/open-dev-tools/test.py]: note the space.

File: nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+<title>Nodejs Modules Debugging</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="author" title="zhang xiaoyux" href="mailto:xiaoyux.zhang@intel.com">
+<link rel="help" href="http://docs.nwjs.io/en/latest/For%20Users/Debugging%20with%20DevTools/">
+</head>
+<body id="demo">
+<ol>
+   <li>'Right Click App'</li>
+   <li>'Select "Inspect Background Page" '</li>
+   <li>'It will is pass if debugging Tools is opened '</li>
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/nodejs-modules-debugging/index.html]: Test passes if Debugging Tools is opened.

File: nwjs-iot/nwjsapi/debugging-with-devTools/remote-debugging/index.html [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+<html>
+<head>
+<meta charset="utf-8">
+<title>Remote Debugging</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="author" title="zhang xiaoyux" href="mailto:xiaoyux.zhang@intel.com">
+<link rel="help" href="http://docs.nwjs.io/en/latest/For%20Users/Debugging%20with%20DevTools/">
+</head>
+<body>
+<p>App run with argument '--remote-debugging-port'</p>
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/remote-debugging/index.html]: Launch App with ....

File: nwjs-iot/nwjsapi/debugging-with-devTools/testcfg.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
+import test
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/testcfg.py]: Add Copyright.
And what is this file used for?

File: nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/build/build.min.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...alignItems:"msFlexAlign",justifyContent:"msFlexPack",order:"msFlexOrder",flexGrow:"msFlexPositive",flexShrink:"msFlexNegative",flexBasis:"msPreferredSize"},a=Object.keys(i).concat("display");n["default"]=function(e,t,n,s){var l=n.browser,u=n.version;return a.indexOf(e)>-1&&("ie_mob"===l||"ie"===l)&&10==u?(delete s[e],o({},i[e]||e,r[t]||t)):void 0},t.exports=n["default"]},{}],37:[function(e,t,n){"use strict";function o(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}Object.defineProperty(n,"__esModule",{value:!0});var r={"space-around":"justify","space-between":"justify","flex-start":"start","flex-end":"end","wrap-reverse":"multiple",wrap:"multiple",flex:"box","inline-flex":"inline-box"},i={alignItems:"WebkitBoxAlign",justifyContent:"WebkitBoxPack",flexWrap:"WebkitBoxLines"},a=Object.keys(i).concat(["alignContent","alignSelf","display","order","flexGrow","flexShrink","flexBasis","flexDirection"]);n["default"]=function(e,t,n){var s=n.browser,l=n.version,u=n.prefix;return a.indexOf(e)>-1&&("firefox"===s&&22>l||"chrome"===s&&21>l||("safari"===s||"ios_saf"===s)&&6.1>=l||"android"===s&&4.4>l||"and_uc"===s)?"flexDirection"===e?{
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/build/build.min.js]: Where does this file come from?

File: nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/css/bundle.css [hunk 13]
```diff
@@ -0,0 +1,24 @@
+/*!
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/css/bundle.css]: Ditto.

File: nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/index.html [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
```
Comment (Honry) [inline nwjs-iot/nwjsapi/debugging-with-devTools/using-devtools-extensions/N-Puzzle/index.html]: Do these files come from upstream?

**Comment by xiaoyu3x** [comment]
I will re-submit PR, so closed it.

**Pull request closed without merging**